src/hb-subset.h

```diff
@@ -84,6 +84,29 @@ typedef enum { /*< flags >*/
   HB_SUBSET_FLAGS_RETAIN_ALL_FEATURES =	     0x00000200u,
 } hb_subset_flags_t;
 
+/**
+ * hb_subset_sets_t:
+ * @HB_SUBSET_SETS_LAYOUT_FEATURE_TAG: the set of layout feature tags that will be retained
+ * in the subset.
+ * @HB_SUBSET_SETS_NAME_INDEX: the set of name ids that will be retained.
+ * @HB_SUBSET_SETS_NAME_LANG_INDEX: the set of name lang ids that will be retained.
+ * @HB_SUBSET_SETS_NO_SUBSET_TABLE_TAG: the set of table tags which specifies tables that should not be
+ * subsetted.
+ * @HB_SUBSET_SETS_DROP_TABLE_TAG: the set of table tags which specifies tables which will be dropped
+ * in the subset.
+ *
+ * List of sets that can be configured on the subset input.
+ *
+ * Since: REPLACE
+ **/
+typedef enum { /*< flags >*/
+  HB_SUBSET_SETS_LAYOUT_FEATURE_TAG = 0,
+  HB_SUBSET_SETS_NAME_INDEX,
+  HB_SUBSET_SETS_NAME_LANG_INDEX,
+  HB_SUBSET_SETS_NO_SUBSET_TABLE_TAG,
+  HB_SUBSET_SETS_DROP_TABLE_TAG,
```

Comment:
Let's add a `HB_SUBSET_SETS_COUNT` at the end? Similarly probably add `HB_SUBSET_FLAG_DEFINED` as or of all.